### PR TITLE
DEVHAS-391: add initial script to test samples against RHTAP

### PR DESCRIPTION
**Pull request by flacatus**
### What does this PR do?:
Create initial ginkgo script reutilizing RHTAP e2e libs to test devfile samples

### Which issue(s) this PR fixes:
_Link to github issue(s)_

### PR acceptance criteria:

- [ ] Contributing guide
_Have you read the [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) and followed its instructions?_
- [x] Test automation
_Does this repository's tests pass with your changes?_
- [ ] Documentation
_Does any documentation need to be updated with your changes?_
- [ ] Check Tools Provider
_Have you tested the changes with existing tools, i.e. Odo, Che, Console? (See [devfile registry contributing guide](https://github.com/devfile/registry/blob/main/CONTRIBUTING.md) on how to test changes)_

### How to test changes / Special notes to the reviewer:

### CODE REVIEW

File: tests/rhtap/types.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	Name        string `yaml:"name"`
+	DisplayName string `yaml:"displayName"`
+	Language    string `yaml:"language"`
+	ProjectType string `yaml:"projectType"`
+	Git         Git    `yaml:"git"`
+}
+
+type Stacks struct {
+	SchemaVersion string    `yaml:"schemaVersion"`
+	Samples       []Samples `yaml:"samples"`
```
Comment (michael-valdron) [inline tests/rhtap/types.go:28]: I think `ExtraDevfileEntries` would be the best descriptor for this entity.

```suggestion
type ExtraDevfileEntries struct {
	SchemaVersion string    `yaml:"schemaVersion"`
	Samples       []SampleEntry `yaml:"samples"`
```

**Comment by michael-valdron** [comment]
Oh also you will need to amend https://github.com/devfile/registry/pull/191/commits/319866f82bc2733b71d707f11050d454cbc819ef to include the signoff to comply with DCO.

File: tests/rhtap/rhtap_test.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+	var fw *testHub.ControllerHub
+	var testNamespace string
+	component := &appservice.Component{}
+	cdq := &appservice.ComponentDetectionQuery{}
+
+	stacks, err := LoadExtraDevfileEntries(stacksFile)
+	Expect(err).NotTo(HaveOccurred())
+
+	for _, stack := range stacks.Samples {
+		stack := stack
```
[the remark this answers is not in the document]
Comment (flacatus) [reply tests/rhtap/rhtap_test.go:43]: It is needed when u run in parallel inside of a describer

**Review by michael-valdron — APPROVED** [review]
/lgtm

**Comment by maysunfaisal** [comment]
Merged after confirming with @flacatus

**Pull request merged**